app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,57 @@
+class SalesController < ApplicationController
+  before_action :set_sale, only: [:edit, :update, :show]
+
+  def index
+    @sales = Sale.all
+  end
+
+  def new
+    @sale = Sale.new
+    @sale.build_vehicle
+    @sale.build_seller
+    @sale.build_purchaser
+  end
+
+  def edit
+  end
+
+  def update
+    @sale.update(sale_params)
+    respond_with @sale
+  end
+
+  def create
+    @sale = SaleCreator.create(sale_params)
+    respond_with @sale
+  end
+
+  def show
+  end
+
+  private
+
+  def set_sale
+    @sale = Sale.find(params[:id])
```

Comment:
I do not recommend this.

=> https://blog.codeminer42.com/controller-best-practices-dont-hide-instance-variables-5e8bf067156